src/API/controllers/user.controller.ts

```diff
@@ -20,17 +20,17 @@ export class UserController {
       if (errors.length) {
         throw new UserError(errors);
       }
-      const user = await this.userService.registerUser(dto);
+      const user = await this.userService.register(dto);
       const token = genToken({
-        id: user.id,
-        email: user.email,
+        userId: user.id,
+        userEmail: user.email,
         isEmailVerified: user.isEmailVerified,
         tokenType: Token.ACCESS,
       });
 
       const emailVerificationToken = genToken({
-        id: user.id,
-        email: user.email,
+        userId: user.id,
+        userEmail: user.email,
```

Comment:
@Saba-Izraqi
 in the middleware, we're checking using `decoded.email`, which doesn't exist in the token payload. This will always result in unauthorized access.
 
```suggestion
        email: user.email,
```

or change it in the middleware